### PR TITLE
feat: lightning coin support

### PR DESCRIPTION
**Pull request by josepfo**

### CODE REVIEW

File: lib/new-admin/services/funding.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ const reflect = p => p.then(value => ({ value, status: 'fulfilled' }), error =>
 
 function getFunding () {
   return settingsLoader.loadLatest().then(settings => {
-    const cryptoCodes = configManager.getAllCryptoCurrencies(settings.config)
+    const cryptoCodes = _.filter(code => coinUtils.getExternalCryptoCode(code) === code)(configManager.getAllCryptoCurrencies(settings.config))
```
Comment (siiky) [inline lib/new-admin/services/funding.js]: Nitpick: unnecessary parens: `s/)(/, /`

File: lib/plugins/wallet/galoy/galoy.js [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+  }
+  return Promise.resolve(true)
+    .then(() => {
+      return axios({
+        method: 'post',
+        url: URI,
+        headers: headers,
+        data: graphqlQuery
+      })
+    })
```
Comment (siiky) [inline lib/plugins/wallet/galoy/galoy.js]: Nitpick: unnecessary `Promise.resolve`. You can just chain the `then()` and `catch()` to the `axios()`:

```js
  return axios({
    method: 'post',
    url: URI,
    headers: headers,
    data: graphqlQuery
  })
  .then(...)
  .catch(...)
```

File: lib/plugins/wallet/galoy/galoy.js [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+              }
+            }
+          }
+        }
+        id
+        phone
+        twoFAEnabled
+        username
+      }
+    }`,
```
Comment (siiky) [inline lib/plugins/wallet/galoy/galoy.js]: Why request fields you don't want (`createdAt`, `id`, `phone`, `twoFAEnabled`, `username`)? It happens with a few of the other queries too.

File: lib/plugins/wallet/galoy/galoy.js [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+    'variables': { 'input': { 'address': `${address}`, 'amount': `${cryptoAtoms}`, 'walletId': `${walletId}` } }
+  }
+  return request(sendOnChain, token)
+    .then(result => {
+      return result.data.onChainPaymentSend
+    })
+    .catch(err => {
+      throw err
+    })
```
Comment (siiky) [inline lib/plugins/wallet/galoy/galoy.js]: Isn't this the same as not `catch()`ing in the first place? There are some other instances of this.

File: lib/plugins/wallet/galoy/galoy.js [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+    .then(() => getGaloyAccount(account.apiKey))
+    .then(galoyAccount => {
+      const wallet = _.head(
+        _.filter(wallet => wallet.walletCurrency === externalCryptoCode &&
+          wallet.id === galoyAccount.defaultWalletId &&
+          wallet.id === account.walletId)(galoyAccount.wallets)
+      )
+      const transactions = wallet.transactions.edges
+      if (isLightning(address)) {
+        const paymentHash = invoice.decode(address).paymentHash.toString('hex')
```
Comment (siiky) [inline lib/plugins/wallet/galoy/galoy.js:282]: Doesn't this `invoice.decode()` ever fail?

**Review by siiky — CHANGES_REQUESTED** [review]

File: lib/plugins/wallet/galoy/galoy.js (hunk 3 above)
Comment (josepfo) [reply lib/plugins/wallet/galoy/galoy.js]: Some leftovers from testing

File: lib/plugins/wallet/galoy/galoy.js (hunk 4 above)
Comment (josepfo) [reply lib/plugins/wallet/galoy/galoy.js]: Initially wanted to throw more informative errors but ended up leaving this as is

File: lib/plugins/wallet/galoy/galoy.js (hunk 5 above)
Comment (josepfo) [reply lib/plugins/wallet/galoy/galoy.js:282]: yes, in that case, the function will throw an error

**Review by siiky — APPROVED** [review]

**Pull request closed without merging**